### ISSUE TITLE
Python bindings

### ISSUE THREAD

**Issue by talonrenaud** [paceholder/nodeeditor#109]
This library is awesome, tested it using Qt but I could really make use of it using Python/Pyside2.
Any chance somebody would be savvy enough to generate python bindings for this ?
Thanks !

**Comment by 22459** [comment]
I second this request. ;)

**Comment by 22459** [comment]
Meanwhile I found this [https://github.com/EricTRocks/pyflowgraph](https://github.com/EricTRocks/pyflowgraph) , I haven't gotten around to trying to use it yet. There is zero documentation (though there is the tests directory with two examples) but apparently it originates from this project: [http://kraken250.rssing.com/chan-66016167/all_p1.html](http://kraken250.rssing.com/chan-66016167/all_p1.html)

**Comment by FMichelD** [comment]
Hello, I started a conversion of this project from C++ to PyQt5.
It is almost a direct translation from c ++ to python, there are still some
bugs but you can already do something.

The port is at https://github.com/FMichelD/PyQt5Nodes

ps. sorry my english.



2017-12-13 22:57 GMT-02:00 22459 <notifications@github.com>:

**Comment by prusnak** [comment]
Using PySide2 `shiboken2` might be a better idea. I would love the nodeeditor to become a standard part of the Qt/PySide2!

**Comment by Togogermini** [comment]
@FMichelD I am finding it difficult to get some of the entry point modules I found in PyQt5Nodes to run. The ones I tried directly from windows command line are PyQt5Nodes/examples/calculator/main.py and PyQt5Nodes/examples/connections_color/main.py. Running them caused the python interpreter to crash. Any ideas how I can get these modules to work? Thanks in advance.

**Comment by FMichelD** [comment]
@Togogermini, you could call the command
PyQt5Nodes/examples/calculator/main.py
from windows command line using:

pyhton PyQt5Nodes/examples/calculator/main.py

and send me the output of the command?

ps. I have to remember to remove PyQt5Nodes / examples / connections_color
/ main.py, this is still not working
*:)*





Em qui, 23 de ago de 2018 às 16:17, Togogermini <notifications@github.com>
escreveu:

**Comment by dridk** [comment]
I would love this binding too !

**Comment by klauer** [comment]
I found that wrapping the C++ nodeeditor library would result in a not-so-Pythonic library and make packaging/releasing quite a pain. Because of that, I decided to start on a pure-Python port based on qtpy:

https://github.com/klauer/qtpynodeeditor

It's by no means in a final state - documentation and such is lacking, and I've got a list of things to refactor and improve. While most things should work, I wouldn't recommend basing too much code off of it just yet.

Also, because it uses qtpy it should be compatible with PySide, though I've only tested with PyQt5 as of yet.
